src/strawman/collections/CollectionStrawMan6.scala

```diff
@@ -0,0 +1,980 @@
+      for (ch <- s) sb += f(ch)
+      sb.result
+    }
+
+    /** Overloaded version of `flatMap` that gives back a string, where the inherited
+     *  version gives back a sequence.
+     */
+    def flatMap(f: Char => String): String = {
+      val sb = new StringBuilder
+      for (ch <- s) sb ++= f(ch)
+      sb.result
+    }
+
+    /** Overloaded version of `++` that gives back a string, where the inherited
+     *  version gives back a sequence.
+     */
+    def ++(xs: IterableOnce[Char]): String = {
+      val sb = new StringBuilder() ++= s
+      for (ch <- xs.iterator) sb += ch
+      sb.result
+    }
+
+    /** Another overloaded version of `++`. */
+    def ++(xs: String): String = s + xs
+  }
+
+  class StringBuilder extends Builder[Char, String] {
+    private val sb = new java.lang.StringBuilder
+
+    def += (x: Char) = { sb.append(x); this }
+
+    /** Overloaded version of `++=` that takes a string */
+    def ++= (s: String) = { sb.append(s); this }
+
+    def result = sb.toString
+
+    override def toString = result
+  }
+
+  case class StringView(s: String) extends IndexedView[Char] {
+    val start = 0
+    val end = s.length
+    def apply(n: Int) = s.charAt(n)
+    override def className = "StringView"
+  }
+
+  /** Decorator to add collection operations to arrays.
+   */
+  implicit class ArrayOps[A](val xs: Array[A])
+  extends AnyVal with IterableOps[A]
+     with SeqMonoTransforms[A, Array[A]]
+     with Buildable[A, Array[A]] {
+
+    protected def coll = new ArrayView(xs)
+    def iterator = coll.iterator
+
+    override def view = new ArrayView(xs)
+
+    def elemTag: ClassTag[A] = ClassTag(xs.getClass.getComponentType)
+
+    protected def fromIterableWithSameElemType(coll: Iterable[A]): Array[A] = coll.toArray[A](elemTag)
+
+    def fromIterable[B: ClassTag](coll: Iterable[B]): Array[B] = coll.toArray[B]
+
+    protected[this] def newBuilder = new ArrayBuffer[A].mapResult(_.toArray(elemTag))
+
+    override def knownSize = xs.length
+
+    override def className = "Array"
+
+    def map[B: ClassTag](f: A => B): Array[B] = fromIterable(View.Map(coll, f))
+    def flatMap[B: ClassTag](f: A => IterableOnce[B]): Array[B] = fromIterable(View.FlatMap(coll, f))
+    def ++[B >: A : ClassTag](xs: IterableOnce[B]): Array[B] = fromIterable(View.Concat(coll, xs))
+    def zip[B: ClassTag](xs: IterableOnce[B]): Array[(A, B)] = fromIterable(View.Zip(coll, xs))
+  }
+
+  case class ArrayView[A](xs: Array[A]) extends IndexedView[A] {
+    val start = 0
+    val end = xs.length
+    def apply(n: Int) = xs(n)
+    override def className = "ArrayView"
+  }
+
+  /* ---------- Views -------------------------------------------------------*/
+
+  /** Concrete collection type: View */
+  trait View[+A] extends Iterable[A] with IterableLike[A, View] {
+    override def view = this
+
+    /** Avoid copying if source collection is already a view. */
+    override def fromIterable[B](c: Iterable[B]): View[B] = c match {
```

Comment:
Could you have types like `IndexedView` with additional operations on views that are based on IndexedSeqs? This approach doesn't look very extensible because the View operation subclasses for that would all have to extend `IndexedView`.